src/tasks/git_remote.rs

```diff
@@ -4,14 +4,89 @@ use tracing_batteries::prelude::*;
 
 pub struct GitRemote<'a> {
     pub name: &'a str,
+    pub host: String,
+    pub namespace: String,
+    pub is_ssh: bool,
+    pub has_dot_git: bool,
 }
 
 impl Default for GitRemote<'static> {
     fn default() -> Self {
-        Self { name: "origin" }
+        Self { name: "origin", host: "".to_string(), namespace: "".to_string(), is_ssh: false, has_dot_git: false }
     }
 }
 
+impl GitRemote<'static> {
+    pub fn parse(url: &str) -> Option<Self> {
+        if url.starts_with("https://") {
+            // Example: https://github.com/org/repo or /org/repo.git
+            let url = url.strip_prefix("https://")?;
+            let mut parts = url.splitn(2, '/');
+            let host = parts.next()?.to_string();
+            let path = parts.next()?;
+
+            let path_parts = path.split('/').collect::<Vec<_>>();
+            if path_parts.len() != 2 {
```

Comment:
Unfortunately this breaks for things like Azure DevOps (where the URL looks something like `https://dev.azure.com/myorg/myproject/_git/myrepo.git`). Still, have a look at my comment on `git::remote` itself, might save you from needing to fight this battle at all.

Reply:
1. I just realised how insanely I have overcomplicated things :D